incydr/cli/cmds/audit_log.py

```diff
@@ -52,49 +73,108 @@ def search(
     event_types: Optional[str],
     resource_ids: Optional[str],
     user_types: Optional[str],
+    checkpoint_name: Optional[str],
 ):
     """
     Search audit log events.  Returns all events that match the search criteria with paging.
 
     Defaults to searching for most recent events.
 
     Results will be output to the console by default, use the `--output` option to send data to a server.
+
+    Checkpointing is available through the --checkpoint <checkpoint-name> option and will only return new results
+    on subsequent queries with that same checkpoint. Checkpointing filters by timestamp, additional filter
+    options will need to be included in each run.
     """
-    # TODO: checkpointing
     client = ctx.obj()
+    cursor = _get_cursor_store(client.settings.api_client_id)
 
-    date_range = DateRange(
-        startTime=None if not start else parse_timestamp_to_posix_timestamp(start),
-        endTime=None if not end else parse_timestamp_to_posix_timestamp(end),
-    )
+    if output:
+        format_ = TableFormat.raw_json
+
+    # skip pydantic modeling when output will just be json
+    if format_ in ("json", "raw-json"):
+
+        def yield_all_events(request_):
+            for page_num in count(0):
+                request_.pageNum = page_num
```

Comment:
Not sure how we didn't catch this before, but the audit log request param should be `page`, not `pageSize`. I found this because I was testing this command and it was just infinitely looping the first page.

Reply:
Also, looks like the page num starts at 0 for the audit log API, but we're starting the `count()` in the client at 1 in places.

Reply:
Gotcha, I'll double check the query params